File: src/test/java/cz/vhromada/catalog/mapper/MusicMapperTest.java
package cz.vhromada.catalog.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Music;
import cz.vhromada.catalog.utils.MusicUtils;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link cz.vhromada.catalog.domain.Music} and {@link Music}.
 *
 * @author Vladimir Hromada
 */
class MusicMapperTest {

    private MusicMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(MusicMapper.class);
    }

    /**
     * Test method for {@link MusicMapper#map(Music)}.
     */
    @Test
    void map() {
        final Music music = MusicUtils.newMusic(1);
        final cz.vhromada.catalog.domain.Music musicDomain = mapper.map(music);

        MusicUtils.assertMusicDeepEquals(music, musicDomain);
    }

    /**
     * Test method for {@link MusicMapper#map(Music)} with null music.
     */
    @Test
    void map_NullMusic() {
        assertThat(mapper.map(null)).isNull();
    }


    /**
     * Test method for {@link MusicMapper#mapBack(cz.vhromada.catalog.domain.Music)}.
     */
    @Test
    void mapBack() {
        final cz.vhromada.catalog.domain.Music musicDomain = MusicUtils.newMusicDomain(1);
        final Music music = mapper.mapBack(musicDomain);

        MusicUtils.assertMusicDeepEquals(music, musicDomain);
    }

    /**
     * Test method for {@link MusicMapper#mapBack(cz.vhromada.catalog.domain.Music)} with null music.
     */
    @Test
    void mapBack_NullMusic() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: src/test/kotlin/cz/vhromada/catalog/mapper/MusicMapperTest.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.CatalogTestConfiguration
import cz.vhromada.catalog.entity.Music
import cz.vhromada.catalog.utils.MusicUtils
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [cz.vhromada.catalog.domain.Music] and [Music].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogTestConfiguration::class])
class MusicMapperTest {

    /**
     * Instance of [MusicMapper]
     */
    @Autowired
    private lateinit var mapper: MusicMapper

    /**
     * Test method for [MusicMapper.map].
     */
    @Test
    fun map() {
        val music = MusicUtils.newMusic(1)
        val musicDomain = mapper.map(music)

        MusicUtils.assertMusicDeepEquals(music, musicDomain)
    }

    /**
     * Test method for [MusicMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val musicDomain = MusicUtils.newMusicDomain(1)
        val music = mapper.mapBack(musicDomain)

        MusicUtils.assertMusicDeepEquals(music, musicDomain)
    }

}
